oap-server/server-receiver-plugin/configuration-discovery-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/recevier/configuration/discovery/AgentConfigurationsWatcher.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.recevier.configuration.discovery;
+
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.skywalking.oap.server.configuration.api.ConfigChangeWatcher;
+import org.apache.skywalking.oap.server.core.Const;
+import org.apache.skywalking.oap.server.library.module.ModuleProvider;
+
+/**
+ * AgentConfigurationsWatcher used to handle dynamic configuration changes.
+ */
+public class AgentConfigurationsWatcher extends ConfigChangeWatcher {
+    private volatile String settingsString;
+    private volatile Map<String, AgentConfigurations> activeAgentConfigurationsCache;
+
+    public AgentConfigurationsWatcher(Map<String, AgentConfigurations> agentConfigurationsCache,
+                                      ModuleProvider provider) {
+        super(ConfigurationDiscoveryModule.NAME, provider, "agentConfigurations");
+        this.settingsString = Const.EMPTY_STRING;
+        this.activeAgentConfigurationsCache = agentConfigurationsCache;
+    }
+
+    @Override
+    public void notify(ConfigChangeEvent value) {
+        if (value.getEventType().equals(EventType.DELETE)) {
+            settingsString = Const.EMPTY_STRING;
+            this.activeAgentConfigurationsCache = new HashMap<>();
+        } else {
+            settingsString = value.getNewValue();
+            AgentConfigurationsReader agentConfigurationsReader =
+                new AgentConfigurationsReader(new StringReader(value.getNewValue()));
+            this.activeAgentConfigurationsCache = agentConfigurationsReader.readAgentConfigurations();
+        }
+    }
+
+    @Override
+    public String value() {
+        return settingsString;
+    }
+
+    public Map<String, AgentConfigurations> getActiveAgentConfigurationsCache() {
```

Comment:
Don't return the Map directly, you should return Option<AgentConfigurations>. And use UUID and serviceName as parameter.

The whole UUID mechanism is missing, please fix this.

Reply:
done.